### PR TITLE
Log all prompts from specific users

### PR DESCRIPTION
**Pull request by rafax**
This PR adds support for logging all prompts sent by specific actors (provided in `CODY_GATEWAY_IDENTIFIERS_TO_LOG_FOR` env var).
Slack [context](https://sourcegraph.slack.com/archives/C05497E9MDW/p1716475547410459).

Closes PRIME-250.

## Test plan

- locally tested
- e2e tests pass

### CODE REVIEW

**Review by taras-yemets — APPROVED** [review]
LGTM

File: cmd/cody-gateway/internal/httpapi/completions/upstream.go [hunk 1]
```diff
@@ -327,6 +334,9 @@ func makeUpstreamHandler[ReqT UpstreamRequest](
 			if flaggingResult.IsFlagged() {
 				requestMetadata = events.MergeMaps(requestMetadata, getFlaggingMetadata(flaggingResult, act))
 			}
+			if act.IsDotComActor() && config.IdentifiersToLogFor.Has(act.ID) {
+				requestMetadata["full_prompt"] = body.BuildPrompt()
+			}
```
Comment (vdavid) [inline cmd/cody-gateway/internal/httpapi/completions/upstream.go:339]: 👀 marking this as the meat of this PR

**Review by vdavid — APPROVED** [review]
Looks good to me, thanks for adding this, Rafal!

**Comment by chenkc805** [comment]
Thanks Rafal. Where is `CODY_GATEWAY_IDENTIFIERS_TO_LOG_FOR` defined?

**Comment by rafax** [comment]
> Thanks Rafal. Where is `CODY_GATEWAY_IDENTIFIERS_TO_LOG_FOR` defined?

The value is set in Cody Gateway configuration in the infrastructure private repo (will follow-up on Slack).

**Pull request merged**